### PR TITLE
Issue 13889 error handling

### PR DESCRIPTION
**Pull request by jgambarios**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java [hunk 1]
```diff
@@ -759,25 +716,21 @@ else if(UtilMethods.isSet(y)){
                     }
                 }else if(field.getFieldType().equals(Field.FieldType.IMAGE.toString()) || field.getFieldType().equals(Field.FieldType.FILE.toString())) {
                     String filePath = value;
-                    if(field.getFieldType().equals(Field.FieldType.IMAGE.toString()) && !UtilMethods.isImage(filePath))
-                    {
+                    if (field.getFieldType().equals(Field.FieldType.IMAGE.toString())
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java]: Field.FieldType.IMAGE.toString()equals(field.getFieldType())

File: dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java [hunk 2]
```diff
@@ -814,46 +766,42 @@ else if(UtilMethods.isSet(y)){
...
-                {
+                if (UtilMethods.isSet(relatedQuery)) {
                     relatedContentlets = conAPI.checkoutWithQuery(relatedQuery, user, false);
 
                     //validate if the contenlet retrieved are from the correct typ
-                    if(FactoryLocator.getRelationshipFactory().isParent(relationship,contentType))
-                    {
-                        for(Contentlet contentlet : relatedContentlets)
-                        {
+                    if (FactoryLocator.getRelationshipFactory()
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java]: double check if this method is closing the db here

File: dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java [hunk 3]
```diff
@@ -1479,6 +1422,73 @@ else if(host != null) {
...
+            throw new DotDataException(e.getMessage(), e);
+        }
+
+        return executeWfAction;
+    }
+
+    /**
+     * Sets a warning message related to the Action associated to the Contentlet
+     */
+    private static void setActionWarning(HashMap<String, List<String>> results,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java]: set to final the arg

File: dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java [hunk 4]
```diff
@@ -1479,6 +1422,73 @@ else if(host != null) {
...
+                .add(LanguageUtil.get(user, "Line--") + " " + lineNumber + ". " +
+                        LanguageUtil.get(user, generalErrorKey)
+                        + specificErrorMessage);
+        // if the user doesn't have access to the action then removed it from
+        // the content to avoid troubles executing the action set on the
+        // dropdown or on the checkin
+        contentlet.setStringProperty(Contentlet.WORKFLOW_ACTION_KEY, null);
+    }
+
+    private static Object validateDateTypes(int lineNumber, Field field, String value,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java]: set to final the arg

File: dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java (hunk 2 above)
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java]: Looks safe

**Pull request merged**